src/main/java/seedu/address/model/ingredient/IngredientQuantity.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.ingredient;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * A class to represent the ingredientQuantity in an ingredient
+ */
+
+public class IngredientQuantity {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ingredient's quantity should be non-zero unsigned integer.";
```

Comment:
still integer? i remember your user guide wanted to allow any decimal amount.

whichever approach you choose, ensure that implementation is consistent with user guide.